src/lib.rs

```diff
@@ -98,21 +108,26 @@ where
     /// # use libwallet::{Wallet, SimpleVault, sr25519, Result};
     /// # #[async_std::main] async fn main() -> Result<()> {
     ///
-    /// let wallet = Wallet::new(SimpleVault::<sr25519::Pair>::new()).unlock("").await?;
+    /// let wallet = Wallet::new(SimpleVault::<sr25519::Pair>::new()).unlock(()).await?;
     /// let signature = wallet.sign(&[0x01, 0x02, 0x03]);
     /// assert!(signature.is_ok());
     /// # Ok(()) }
     /// ```
     pub fn sign(&self, message: &[u8]) -> Result<SignatureOf<V, C>> {
-        Ok(self.root_account()?.sign(message))
+        match self.root_account()?.sign(message) {
```

Comment:
Changed in commit https://github.com/virto-network/libwallet/pull/15/commits/c9c87195f2b85e057a462597afb8e8432856542c
However, the error here isn't that the wallet is locket, but that there was a problem getting the pair of the account. This changed because a sub account may return an error if the derivation fails, but this won't happen with the root account.